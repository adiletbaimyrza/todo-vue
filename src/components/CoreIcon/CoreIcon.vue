<template>
  <img
    :class="$style['icon']"
    :src="src"
    :width="width"
    :height="height"
    alt="Icon"
  />
</template>

<script>
export default {
  name: 'CoreIcon',
  props: {
    src: {
      type: String,
      required: true,
    },
    width: {
      type: String,
      default: '24px',
    },
    height: {
      type: String,
      default: '24px',
    },
  },
}
</script>

<style lang="scss" module>
.icon {
  display: block;
}
</style>
