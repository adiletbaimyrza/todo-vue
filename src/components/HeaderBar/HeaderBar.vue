<template>
  <header :class="$style['bar']">
    <h1 :class="$style['bar__heading']">todo</h1>
    <button type="button" :class="$style['bar__button']">
      <CoreIcon
        :src="require('../../assets/moon.svg')"
        height="32px"
        width="32px"
      />
    </button>
  </header>
</template>

<script>
import CoreIcon from '../CoreIcon/CoreIcon.vue'

export default {
  name: 'HeaderBar',
  components: {
    CoreIcon,
  },
}
</script>

<style lang="scss" module>
.bar {
  display: flex;
  justify-content: space-between;
  width: 600px;
  margin-bottom: 4em;
  margin-top: 16vh;

  &__heading {
    text-transform: uppercase;
    color: #ffffff;
    letter-spacing: 0.5em;
    cursor: default;
  }
}
</style>
