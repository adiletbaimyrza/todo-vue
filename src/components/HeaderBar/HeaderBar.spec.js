import { shallowMount } from '@vue/test-utils'
import HeaderBar from './HeaderBar.vue'
import CoreIcon from '../CoreIcon/CoreIcon.vue'

/* global jest describe beforeEach it expect */
describe('Header.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(HeaderBar)
  })

  it('renders correctly', () => {
    expect(wrapper.exists()).toBe(true)
  })

  it('renders todo text correctly', () => {
    const text = 'todo'
    const renderedText = wrapper.text()
    expect(renderedText).toContain(text)
  })

  it('renders CoreIcon correctly', () => {
    const coreIcon = wrapper.findComponent(CoreIcon)
    expect(coreIcon.exists()).toBe(true)
  })

  it('fires toggleTheme method when button is clicked', async () => {
    const toggleThemeMock = jest.fn()
    wrapper.vm.toggleTheme = toggleThemeMock
    const button = wrapper.find('button')
    await button.trigger('click')
    expect(toggleThemeMock).toHaveBeenCalled()
  })
})
