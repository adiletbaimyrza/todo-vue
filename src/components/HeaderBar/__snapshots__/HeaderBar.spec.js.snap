// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBar.vue renders correctly 1`] = `
<header
  class="bar"
  data-testid="HeaderBar"
>
  <h1
    class="bar__heading"
    data-testid="Heading"
  >
     todo 
  </h1>
  <button
    class=""
    data-testid="Toggle"
    type="button"
  >
    <img
      alt="Icon"
      class="icon bar__button__icon"
      data-testid="CoreIcon"
      height="32px"
      src=""
      width="32px"
    />
  </button>
</header>
`;
